test/SimulationCSharpClient.Tests/HttpRetryHandler/HttpRetryMessageHandlerTests.cs

```diff
@@ -54,5 +54,63 @@ public void HttpGet_Success_In_FirstTry()
             // Assert
             Assert.AreEqual(((TestHttpRetryMessageHandlerSuccess)handler.InnerHandler).AttemptedTries, 1);
         }
+
+        [Test]
+        public void HttpRetryMessageHandler_ThrowsException_If_Cancellation_Requested()
+        {
+            // Arrange
+            int maxReTries = 3;
+            var cancellationTokenSource = new CancellationTokenSource();
+            var httpRequestMessage = new HttpRequestMessage(HttpMethod.Get, "http://localhostTestApi.com");
+
+            var handler = new HttpRetryMessageHandler(maxReTries)
+            {
+                InnerHandler = new TestHttpRetryMessageHandlerFailure()
+            };
+
+            var invoker = new HttpMessageInvoker(handler);
+
+            // Act
+            cancellationTokenSource.Cancel();
+            var exception = Assert.Catch<Exception>(() =>
```

Comment:
do you want to test for a specific exception type?

Reply:
changed to verify that  exception is OperationCanceledException in case of Aggregation exception also